### PR TITLE
fix: Only add sidxMapping on successful sidx request and parse.

### PR DESCRIPTION
**Pull request by brandonocasey**
## Description
Requires https://github.com/videojs/mpd-parser/pull/123

Fixes our `Angel One - Widevine, fmp4, webm, subs, alternate audio tracks` and i'm sure it fixes others.

### CODE REVIEW

File: src/dash-playlist-loader.js [hunk 1]
```diff
@@ -112,18 +113,6 @@ export const parseMasterXml = ({ masterXml, srcUrl, clientOffset, sidxMapping })
   return master;
 };
 
-export const generateSidxKey = (sidxInfo) => {
```
Comment (brandonocasey) [inline src/dash-playlist-loader.js:115]: This function already existed in mpd-parser, now that we export it, use that one.

File: src/dash-playlist-loader.js [hunk 2]
```diff
@@ -358,6 +347,7 @@ export default class DashPlaylistLoader extends EventTarget {
 
     const fin = (err, request) => {
       if (this.requestErrored_(err, request, startingState)) {
+        delete sidxMapping[sidxKey];
```
Comment (brandonocasey) [inline src/dash-playlist-loader.js:350]: This is the fix

File: src/dash-playlist-loader.js [hunk 3]
```diff
@@ -350,20 +339,28 @@ export default class DashPlaylistLoader extends EventTarget {
...
+      let sidx;
 
-      sidxMapping[sidxKey].sidx = sidx;
+      try {
+        sidx = parseSidx(toUint8(request.response).subarray(8));
+      } catch (e) {
+        // sidx parsing failed.
+        this.requestErrored_(e, request, startingState);
+        return;
+
```
Comment (gkatsev) [inline src/dash-playlist-loader.js:357]:
```suggestion
```

**Review by gkatsev — APPROVED** [review]

**Review by gkatsev — APPROVED** [review]
I think we can merge it with the 0.03% reduction in coverage.

**Pull request merged**